### DISCUSSION TITLE
Alternative firmwares (specifically, MicroPython)

### DISCUSSION THREAD

**Discussion by andypiper** [Phambili-Tech/Newt_Display#26] [Ideas]
I'm going to start noodling at a MicroPython build for the Newt, and wondered whether others in the community are interested, or whether I'm an audience of 1 🙂 

There are a couple of key components that need drivers, one being the RV3028 RTC chip. I've identified a range of other Python or MicroPython drivers for the chip, and I've also ordered a few different breakout boards for the RV3028, so that I can test whatever I build across devices to confirm I'm getting the protocol correct.

I'll update as I make any progress, this is very much a background task / side project on my part, but wanted to let the community know that this is something I'm interested in, in case anyone else is going down a similar path of exploration.

**Comment by tjfoth** [comment]
Count me in!

---
Tom Foth
***@***.***
***@***.***

**Comment by mattytrentini** [comment]
I'm sure you'll be unsurprised to discover that I'm keen too @andypiper!

**Reply by andypiper** [reply]
Absolutely 🙂 

I have got as far as putting MicroPython onto my spare Newt, but for the most part not done any packaging around modules yet. I had a play with a driver for the Sharp display, and I also have one of each of the 4 or 5 RV3028 modules I could find from various vendors, to start working on increasing function coverage of the driver(s) there. And then I got diverted onto other things 😄 - I'm sure I'll be coming back along this road shortly!
